lib/puma/configuration.rb

```diff
@@ -167,14 +169,24 @@ def flatten!
       self
     end
 
+    def default_workers
+      return 0 if Puma.jruby? || Puma.windows?
+      1
+    end
+
+    def default_max_threads
+      return 5 if RUBY_ENGINE.nil? || RUBY_ENGINE == 'ruby'
```

Comment:
Hmmm...is this is best way to detect MRI?

Also, we should abstract this behind `Puma.mri?`